util/slice.cc

```diff
@@ -60,19 +75,19 @@ class FixedPrefixTransform : public SliceTransform {
 class CappedPrefixTransform : public SliceTransform {
  private:
   size_t cap_len_;
-  std::string name_;
 
  public:
-  explicit CappedPrefixTransform(size_t cap_len)
-      : cap_len_(cap_len),
-        // Note that if any part of the name format changes, it will require
-        // changes on options_helper in order to make RocksDBOptionsParser work
-        // for the new change.
-        // TODO(yhchiang): move serialization / deserializaion code inside
-        // the class implementation itself.
-        name_("rocksdb.CappedPrefix." + ToString(cap_len_)) {}
+  explicit CappedPrefixTransform(size_t cap_len) : cap_len_(cap_len) {
+    RegisterOptions(Name(), &cap_len_, &slice_transform_length_info);
+  }
 
-  const char* Name() const override { return name_.c_str(); }
```

Comment:
Is there any backward compatibility issue when Name() is used here. In the legacy code, we will get "rocksdb.CappedPrefix.(cap_len)" But now, we only have `rocksdb.CappedPrefix`. I'm not sure if user will change them to `GetId()`

Reply:
I do not think this will be an issue.  First, I believe the Name() method was primarily used for Dump and related functions.  I did miss one place in Dump that I updated and will update HISTORY with this change.